### PR TITLE
 Add tests for single-byte read/write methods

### PR DESCRIPTION
**Pull request by dfunckt**
Also fixes minor linting issues.

Will update the upstream PR as well after this gets merged.

### CODE REVIEW

File: circbuf_test.go [hunk 1]
```diff
@@ -196,3 +196,46 @@ func TestBuffer_Reset(t *testing.T) {
...
+
+func TestBuffer_ReadByte(t *testing.T) {
+	inp := []byte("hello world")
+
+	buf, err := NewBuffer(int64(len(inp)))
+	if err != nil {
+		t.Fatalf("err: %v", err)
+	}
+
+	if _, err := buf.Write([]byte("hell")); err != nil {
```
Comment (zozo123) [inline circbuf_test.go:227]: Is it possible to be more descriptive in the errors prints?

**Review by zozo123 — COMMENTED** [review]
LGTM, try to be more descriptive in your error messages.

File: circbuf_test.go (hunk 1 above)
Comment (dfunckt) [reply circbuf_test.go:227]: @zozo123 not sure what you mean -- which error prints?

File: circbuf_test.go [hunk 2]
```diff
@@ -196,3 +196,46 @@ func TestBuffer_Reset(t *testing.T) {
...
+		t.Fatalf("bad: %v", actual)
+	}
+}
+
+func TestBuffer_ReadByte(t *testing.T) {
+	inp := []byte("hello world")
+
+	buf, err := NewBuffer(int64(len(inp)))
+	if err != nil {
+		t.Fatalf("err: %v", err)
```
Comment (zozo123) [inline circbuf_test.go:224]: I meant this kind of error. Make it more informative. `(failed to allocate new buffer %v, err)` something like that.

File: circbuf_test.go (hunk 1 above)
Comment (zozo123) [reply circbuf_test.go:227]: see above. ignore this comment

File: circbuf_test.go (hunk 2 above)
Comment (dfunckt) [reply circbuf_test.go:224]: @zozo123 I'm just following the style of the rest of the document -- if you look around `circbuf_test.go` you'll see it's `err:` and `bad:` everywhere.

**Comment by dfunckt** [comment]
@zozo123 in reply to your comment at https://github.com/resin-os/circbuf/commit/fb40b1368ec87ab6c26f2682f31b91d8f362c655#commitcomment-25757550 -- I updated the code based on golint's output, you might want to have another look.

BTW -- can you please add your comments *on* the PR? I keep hunting around to find them.

**Review by petrosagg — APPROVED** [review]

**Pull request merged**